packages/test-renderer/src/index.tsx

```diff
@@ -34,67 +34,55 @@ const create = async (element: React.ReactNode, options?: Partial<CreateOptions>
     },
   })
 
-  const _fiber = canvas
+  const _root = createRoot(canvas).configure({ frameloop: 'never', ...options, events: undefined })
+  const _store = mockRoots.get(canvas)!.store
 
-  const _root = createRoot(_fiber).configure({ frameloop: 'never', ...options, events: undefined })
-
-  let scene: MockScene = null!
-
-  await act(async () => {
-    scene = _root.render(element).getState().scene as unknown as MockScene
-  })
-
-  const _store = mockRoots.get(_fiber)!.store
+  await act(async () => _root.render(element))
+  const scene = (_store.getState().scene as any).__r3f as MockInstance<THREE.Scene>
 
   return {
     scene: wrapFiber(scene),
-    unmount: async () => {
+    async unmount() {
       await act(async () => {
         _root.unmount()
       })
     },
-    getInstance: () => {
+    getInstance() {
       // this is our root
-      const fiber = mockRoots.get(_fiber)?.fiber
-      const current = fiber?.current.child.child
-      if (current) {
-        const root = {
-          /**
-           * we wrap our child in a Provider component
-           * and context.Provider, so do a little
-           * artificial dive to get round this and
-           * pass context.Provider as if it was the
-           * actual react root
-           */
-          current,
-        }
-
-        /**
-         * so this actually returns the instance
-         * the user has passed through as a Fiber
-         */
-        return reconciler.getPublicRootInstance(root)
-      } else {
-        return null
-      }
+      const fiber = mockRoots.get(canvas)?.fiber
+      const current = fiber?.current?.child?.child?.child?.child
+      if (!current) return null
+
+      /**
+       * we wrap our child in a Provider component
+       * and context.Provider, so do a little
+       * artificial dive to get round this and
+       * pass context.Provider as if it was the
+       * actual react root
+       */
+      const root = { current }
+
+      /**
+       * so this actually returns the instance
+       * the user has passed through as a Fiber
+       */
+      return reconciler.getPublicRootInstance(root)
     },
-    update: async (newElement: React.ReactNode) => {
-      const fiber = mockRoots.get(_fiber)?.fiber
-      if (fiber) {
-        await act(async () => {
-          reconciler.updateContainer(newElement, fiber, null, () => null)
-        })
-      }
-      return
+    async update(newElement: React.ReactNode) {
+      if (!mockRoots.has(canvas)) return console.warn('RTTR: attempted to update an unmounted root!')
+
+      await act(async () => {
+        _root.render(newElement)
+      })
```

Comment:
We no longer implement container methods, appending into `<primitive object={scene} />` as the host instance.